src/ng/directive/ngOptions.js

```diff
@@ -449,12 +449,12 @@ var ngOptionsDirective = ['$compile', '$document', '$parse', function($compile,
       if (!multiple) {
 
         selectCtrl.writeValue = function writeNgOptionsValue(value) {
-          var selectedOption = options.selectValueMap[selectElement.val()];
+          var selectedOption = selectElement[0].options[selectElement[0].selectedIndex];
```

Comment:
Hmm, I was just comparing it to the previous code which used `options.selectValueMap`. Could you explain the change?